src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -125,12 +129,42 @@ private FedoraResource locateResource(final Path path, final Session session) {
 
         if (exists.test(path) || path.isRoot()) {
             LOGGER.debug("findRolesForPath: {}", path.getString());
-            return nodeService.find(session, path.toString());
+            final FedoraResource resource = nodeService.find(session, path.toString());
+
+            if (resource.hasType("nt:version")) {
+                LOGGER.debug("{} is a version, getting the baseVersion", resource);
+                return getBaseVersion(resource);
+            }
+            return resource;
         }
         LOGGER.trace("Path: {} does not exist, checking parent", path.getString());
         return locateResource(path.getParent(), session);
     }
 
+    /**
+     * Get the versionable FedoraResource for this version resource
+     *
+     * @param resource the Version resource
+     * @return the base versionable resource or the version if not found.
+     */
+    private FedoraResource getBaseVersion(final FedoraResource resource) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        try {
+            final VersionHistory base = ((Version) resource.getNode()).getContainingHistory();
+            if (base.hasProperty("jcr:versionableUuid")) {
+                LOGGER.debug("versionableUuid : {}", base.getProperty("jcr:versionableUuid").getValue().getString());
```

Comment:
You could probably create a local variable that holds: `base.getProperty(JCR_VERSIONABLE_UUID).getValue().getString())`